source/AI.cpp

```diff
@@ -3296,10 +3295,15 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 						continue;
 					
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double b;
+					if(Preferences::Has("Boarding target priority")) {
```

Comment:
Since the preference isn't going to change inside of the loop, we can move it outside of the loop.